test/tests/helper.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+
+const app = require('../../server');
+const chai = require('chai');
+const db = require('../../db');
+const mongoose = require('mongoose');
+const clearDB  = require('mocha-mongoose')(db.uri);
+const request = require('supertest');
+
+chai.use(require('chai-as-promised'));
+global.expect = chai.expect();
+global.should = chai.should();
+global.request = request(app);
+
+
+
+beforeEach((done, err) => {
+  if(mongoose.connection.db) return done(err);
+
+  mongoose.connect(dbURI);
+  clearDB((err) => {
+    if(err) return err;
+    done();
+  });
+});
+
+afterEach((done) => {
+  mongoose.disconnect();
```

Comment:
Is `mongoose.disconnect()` async or sync?

Reply:
It is async: http://mongoosejs.com/docs/api.html#index_Mongoose-disconnect

Thus you need to wait for it to callback before invoking `done()`
```
afterEach((done) => {
  mongoose.disconnect(done);
});
```